lib/http_requests.py

```diff
@@ -129,41 +139,29 @@ def get_advisories(branch_name):
     advisory_data = []
     for version in yml_data:
         if version in seen:
-            continue  # If we have already included the advisories of a version
-
-        try:
-            _ = yml_data[version]['assembly']['basis']
-        except KeyError:
-            continue  # To make sure assembly and basis keys are present. Eg art3171 in 4.7 does not have those.
-
-        brew_event_id = get_brew_event_id(yml_data[version])
-
-        has_advisories = False  # flag to check if a child that inherits has advisories, in case that has precedence
-        recursion_depth = 0  # counter to prevent infinite loops
-        while not brew_event_id:
-            recursion_depth += 1
-            if recursion_depth == 1000:  # Set maximum recursion depth
-                break
-
-            advisories = get_particular_advisory(yml_data[version])
-            if advisories:
-                advisory_data.append([version, advisories])
-                has_advisories = True
-                break  # exit if advisories exist even though child inherits
-            else:
-                version = yml_data[version]['assembly']['basis']['assembly']  # get the parent version number
-                brew_event_id = get_brew_event_id(yml_data[version])  # update the brew_event id
-
-        if version not in seen:
-            seen[version] = 1  # add version to seen as we have processed the advisories
-
-        if has_advisories:
-            continue  # continue if inherited child has advisories, which is already processed in the while loop
-
-        advisories = get_particular_advisory(yml_data[version])
-        if advisories:
-            jira_link = get_jira_link(yml_data[version])
-            advisory_data.append([version, advisories, jira_link])
+            continue
+        
+        current_advisories, jira_link = process_version_advisories(version, yml_data, seen)
+        if not current_advisories:
+            current_advisories = {}
+
+        # Handle inheritance and overrides
+        basis_version = yml_data[version].get('assembly', {}).get('basis', {}).get('assembly')
+        if basis_version:
+            # Inherit advisories from the basis version if available
+            basis_advisories, _ = process_version_advisories(basis_version, yml_data, seen)
+            if basis_advisories:
+                current_advisories.update(basis_advisories)
```

Comment:
Inheritance could go multi levels. Like for example let say we create `4.14.1`, then something happens and we re promote to `4.14.2` and then something happens again, and we re promote to `4.14.3`. The odds of that happening is low, of course, but we would be future proofing this piece of code. So my suggestion would be to convert this into a while loop and set the `MAX_DEPTH` to 3. So in the future, if we only need to change the constant, if needed.

Reply:
Wasn't thinking of such context, makes sense for me - thank you.